### PR TITLE
Feature/api setup

### PR DESCRIPTION
**Pull request by bmijanovic**
Database is local, so connection string is specific for my computer.

### CODE REVIEW

**Comment by bmijanovic** [comment]
If my API setup works, delete API folder with old one

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,400 @@
...
+
+# Windows Installer files from build outputs
+*.cab
+*.msi
+*.msix
+*.msm
+*.msp
+
+# JetBrains Rider
+*.sln.iml
```
Comment (drmarjanovic) [inline .gitignore:400]: I prefer a new line at the end of the files.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,400 @@
 *.bak
 *.DS_Store
```
Comment (drmarjanovic) [inline .gitignore:2]: You can check the "global gitignore" config. Some info can be found [here](https://stackoverflow.com/questions/7335420/global-git-ignore). 

I use it for adding system and IDE-specific files there. The reason for that - maybe I use IntelliJ (.idea) should be ignored, and someone uses Eclipse... And I don't won't to keep it in project's `.gitignore`.

File: APP/API/obj/Debug/net6.0/API.GeneratedMSBuildEditorConfig.editorconfig [hunk 3]
```diff
@@ -8,9 +8,9 @@ build_property.PlatformNeutralAssembly =
 build_property._SupportedPlatformList = Linux,macOS,Windows
 build_property.RootNamespace = API
 build_property.RootNamespace = API
```
Comment (drmarjanovic) [inline APP/API/obj/Debug/net6.0/API.GeneratedMSBuildEditorConfig.editorconfig]: Is this duplicated?

File: APP/API/obj/Debug/net6.0/API.GeneratedMSBuildEditorConfig.editorconfig [hunk 4]
```diff
@@ -8,9 +8,9 @@ build_property.PlatformNeutralAssembly =
...
-build_property.ProjectDir = G:\FTN\SIMS\ElectricChargers\APP\API\
+build_property.ProjectDir = D:\FTN\IV semestar\SIMS\Projekat\ElectricChargers\APP\API\
```
Comment (drmarjanovic) [inline APP/API/obj/Debug/net6.0/API.GeneratedMSBuildEditorConfig.editorconfig]: Are you sure that `obj/Debug` directory is necessary to be committed?

File: APP/Back/EVChargersAPI/Data/Entities/User.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+using System.Threading.Tasks;
+
+namespace Data.Entities
+{
+    [Table("Person")]
+    public class User
+    {
+        [Column("id")]
+        [JsonPropertyName("id")]
+        public Guid? Id { get; set; }
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/Data/Entities/User.cs:16]: Do they have `Uuid` type? What's the difference?

File: APP/Back/EVChargersAPI/EVChargersAPI.sln [hunk 6]
```diff
@@ -0,0 +1,31 @@
+﻿
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/EVChargersAPI.sln:1]: Is this file necessary?

File: APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Controllers/UserController.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            _userService = userService;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<User>>> GetAll()
+        {
+            IEnumerable<User> users = await _userService.GetAll();
+            return Ok(users);
+        }
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Controllers/UserController.cs:23]: I would add a new line here.

File: APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Repositories/IRepository.cs [hunk 8]
```diff
@@ -0,0 +1,10 @@
+﻿namespace EVChargersAPI.UserManagement.Repositories
+{
+    public interface IRepository<T> where T : class
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Repositories/IRepository.cs:3]: Is `T` convention in .NET or you're using it from Java?

File: APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Repositories/IRepository.cs [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+    {
+        Task<IEnumerable<T>> GetAll();
+        T Create(T item);
+
+        void Save();
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Repositories/IRepository.cs:7]: What's the difference between `Save()` and `Create(T item)`?

File: APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Repositories/IRepository.cs [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+    }
+}
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Repositories/IRepository.cs]: Missing a new line.

File: APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Services/IService.cs [hunk 11]
```diff
@@ -0,0 +1,8 @@
+﻿namespace EVChargersAPI.UserManagement.Services
+{
+    public interface IService<T> where T : class
+    {
+        Task<IEnumerable<T>> GetAll();
+        Task<T> Create(T item);
+    }
+}
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Services/IService.cs]: Missing new line here.

File: APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Repositories/UserRepository.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
+﻿using Data.Context;
+using Data.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace EVChargersAPI.UserManagement.Repositories
+{
+    public interface IUserRepository : IRepository<User>
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Repositories/UserRepository.cs:8]: Are you going to have more `IUserRepository` implementations?

File: APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Controllers/UserController.cs [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        [HttpPost]
+        public async Task<ActionResult<User>> Add(User user)
+        {
+            User createdUser = await _userService.Create(user);
```
Comment (drmarjanovic) [inline APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Controllers/UserController.cs:27]: I'm not familiar with .NET, but I would say this is a similar concept to Scala's language. 

I've investigated a little bit - can you check [this](https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/chaining-tasks-by-using-continuation-tasks) article? Also, [this](https://stackoverflow.com/questions/38660747/c-sharp-transform-an-async-task-from-one-type-to-another) was my entry point to discussion.
